### PR TITLE
Themes: Add activate and customize buttons on theme upload completion

### PR DESCRIPTION
**Pull request by seear**
screen shot 2016-11-30 at 22 40 52

**To Test**
* Needs a .org site with Jetpack > 4.4
* Go to http://calypso.localhost:3000/design/upload/{jetpack site}
* Select or drag a theme zip to upload

**Expected**
* On completion of upload, you should see theme details with new buttons underneath
* Clicking the 'Activate' button should show the 'thanks' modal and activate the theme on the site
* Clicking on 'Try & Customize' should take you to the customizer

The customizer flow is not ideal at the moment—clicking on the X in the top-left brings you back to the empty upload page. We can iterate on this.

### CODE REVIEW

**Comment by ockham** [comment]
I keep getting `Upload problem: Zip file too large to upload.`, so I can't seem to test this right now 😕

**Comment by seear** [comment]
> I keep getting Upload problem: Zip file too large to upload., so I can't seem to test this right now 

What size zip are you using? And is it your sandbox site?

I think the limit here is with the .org site, but I need to check that.

**Comment by seear** [comment]
@ockham interested in your opinion on the option connecting stuff. Can it be made simpler?

File: client/my-sites/themes/theme-upload/index.jsx [hunk 1]
```diff
@@ -203,12 +227,29 @@ class Upload extends React.Component {
 	}
 }
 
+const ConnectedUpload = connectOptions(
+	( props ) => (
+		<Upload { ...props } />
+	)
+);
```
Comment (ockham) [inline client/my-sites/themes/theme-upload/index.jsx]: I think `( props ) => <Component { ... props } />` is essentially the identity HoC, so you can probably shorten this to

```js
const ConnectedUpload = connectOptions( Upload )
```

**Comment by ockham** [comment]
> @ockham interested in your opinion on the option connecting stuff. Can it be made simpler?

That's as simple as it gets, I'm afraid. Or, to put it differently: you've used it the way it's meant to be used 😄

**Comment by ockham** [comment]
> > I keep getting Upload problem: Zip file too large to upload., so I can't seem to test this right now

> What size zip are you using? And is it your sandbox site?

< 3 MB, and yes.

**Comment by seear** [comment]
> That's as simple as it gets, I'm afraid. Or, to put it differently: you've used it the way it's meant to be used 😄

Thanks! 😄

**Comment by seear** [comment]
> < 3 MB, and yes.

Bigger than 2MB? Download something smaller—I think the sandbox sites may have a 2MB limit by default.

**Comment by ockham** [comment]
Okay, tried a 600 kB one, which at first gave me a different error. Remembered then that I hadn't updated my Jetpack to 4.4 yet 😳 . Anyway, works like a charm, and looks beautiful. It's a joy to watch the progress bar and then see the preview thing 😄

File: client/my-sites/themes/theme-upload/index.jsx [hunk 2]
```diff
@@ -184,14 +198,24 @@ class Upload extends React.Component {
...
 
 	render() {
-		const { translate, inProgress, complete, failed } = this.props;
+		const { translate, inProgress, complete, failed, siteId, selectedSite } = this.props;
 		return (
 			<Main>
+				<QueryActiveTheme siteId={ siteId } />
+				<ThanksModal
+					site={ selectedSite }
+					source={ 'details' } />
```
Comment (ockham) [inline client/my-sites/themes/theme-upload/index.jsx]: With a string value like this, can be shortened to `source="details"`. But maybe change `source` to `upload`?

File: client/my-sites/themes/theme-upload/index.jsx [hunk 3]
```diff
@@ -184,14 +198,24 @@ class Upload extends React.Component {
...
 	}
 
 	render() {
-		const { translate, inProgress, complete, failed } = this.props;
+		const { translate, inProgress, complete, failed, siteId, selectedSite } = this.props;
 		return (
 			<Main>
+				<QueryActiveTheme siteId={ siteId } />
+				<ThanksModal
+					site={ selectedSite }
```
Comment (ockham) [inline client/my-sites/themes/theme-upload/index.jsx]: We need to change `ThanksModal` to accept a `siteId` prop at some point 😕

**Comment by ockham** [comment]
Code looks good 👍  Couple of non-blocking comments.

It got me thinking about the sorry state of data fetching, see https://github.com/Automattic/wp-calypso/pull/9744#discussion_r90744147. Maybe we should indeed just move query components where corresponding selectors are, e.g. `<QueryActiveTheme />` to `<ThanksModal />`.

(Unrelated, suggestion for a subsequent PR) Hey or what about storing the `themeId` instead of just `true` in the `completedActivationRequests` reducer @budzanowski? It's part of the `THEME_ACTIVATE_REQUEST_SUCCESS` anyway. That way, we wouldn't need `getActiveTheme()`  inside `<ThanksModal />` (and the `<QueryActiveTheme />` component here) and could just use a return value from `hasActivatedTheme()`. That way, that reducer would finally make some more sense 😄

**Comment by ockham** [comment]
> Remembered then that I hadn't updated my Jetpack to 4.4 yet 😳.

Consequently filed #9804 😄

**Comment by folletto** [comment]
> Anyway, works like a charm, and looks beautiful. It's a joy to watch the progress bar and then see the preview thing 

Agreed. Design wise I'll chime in for some changes later, but the buttons are 👍
